### PR TITLE
Integrate JourneyData and remove old dummy api

### PR DESCRIPTION
**Pull request by samgiles**
Integrate the journey-data library replacing the mock data sources we have at the moment.

This also sets up Realm in Android and iOS (this is untested at the moment).

r? @gmarty

### CODE REVIEW

File: js/scenes/Item.js [hunk 1]
```diff
@@ -70,7 +70,7 @@ class Item extends Component {
 }
 
 Item.propTypes = {
-  itemId: PropTypes.string,
+  itemId: PropTypes.number,
```
Comment (gmarty) [inline js/scenes/Item.js:73]: 👍

**Review by gmarty — APPROVED** [review]
I have nothing to say. It looks really good to me and I like it.

Do we have the API of `journey-data` documented somewhere?

I think it makes sense for @arcturus and @wilsonpage to take a look too.

**Comment by samgiles** [comment]
@gmarty Emerging documentation here: https://github.com/mozilla-magnet/journey-data#quick-start

Gonna make one more change in this PR as I've just changed realm to a peer dependency in journey-data, (so I don't need to install react for the server).

File: .gitignore [hunk 2]
```diff
@@ -53,3 +53,6 @@ fastlane/Preview.html
 fastlane/screenshots
 
 TODO.md
+
+realm-object-server/
+default.realm*
```
Comment (wilsonpage) [inline .gitignore]: A comment to say what these files are and how they will be generated would be handy :)

**Comment by wilsonpage** [comment]
👍 Nice Sir Samuel!

**Pull request merged**